src/shared/machine-id-setup.c

```diff
@@ -48,6 +49,19 @@ static int acquire_machine_id_from_credential(sd_id128_t *ret) {
         return 0;
 }
 
+static bool machine_id_firmware(void) {
+        _cleanup_free_ char *buf = NULL;
+        int r;
+
+        r = proc_cmdline_get_key("systemd.machine_id", PROC_CMDLINE_VALUE_OPTIONAL, &buf);
```

Comment:
also check from credentials if it is not set here with `acquire_machine_id_from_credential()`

Reply:
actually, this needs to be done in generate_machine_id